dothething/dothething/ClipsView.swift

```diff
@@ -12,20 +12,19 @@ struct ClipView: View {
     let clip: Clip
 
     var body: some View {
-        // thumbnail of video
-        Image(uiImage: clip.thumbnail)
-            .resizable()
-            .frame(width: 108, height: 192)
-            .cornerRadius(8)
-            .onTapGesture {
-                print("Tapped \(clip.url)")
-                Thinger.playVideo(videoUrl: clip.url)
+        GeometryReader { geometry in
+            if clip.thumbnail == UIImage() {
+                Image("Placeholder")
+                    .resizable()
+            } else {
+                Image(uiImage: clip.thumbnail)
+                    .resizable()
+                    .onTapGesture {
+                        print("Tapped \(clip.url)")
+                        Thinger.playVideo(videoUrl: clip.url)
+                    }
             }
```

Comment:
The new code introduces a GeometryReader to handle the layout of the clip thumbnails, which is a good approach to ensure the thumbnails are sized correctly based on the available space. The conditional check for the clip's thumbnail being empty and the use of the "Placeholder" image is a sensible way to handle cases where the thumbnail is not yet available.